### PR TITLE
Fixes to enable executing matmul on GPU.

### PR DESCRIPTION
**Pull request by MaheshRavishankar**
The GPU path needs to 
1) Use the pass manager that works on IREE::HAL::ExecutableTargetOp, so that the LinalgTileAndFuse pass can be run on this op. This allows the pass to set the workgroup count region on the IREE::HAL::EntryPointOp.
2) The `linalg.copy` operation introduced by bufferize pass needs to have the right marker so that the ConvertToGPU pass can distribute the op to within threads in a thread block.

This allows execution of matmul on GPU with the linalg on tensors path.

### CODE REVIEW

File: iree/compiler/Conversion/CodegenUtils/MarkerUtils.h [hunk 1]
```diff
@@ -57,6 +57,9 @@ StringRef getVectorizeMarker();
 /// a marker that can be used later to delete these operations.
 StringRef getDeleteMarker();
 
+/// Returns the marker set on an operation, or "" if no marker is set
```
Comment (antiagainst) [inline iree/compiler/Conversion/CodegenUtils/MarkerUtils.h:60]: Nit: dot to end the sentence to be consistent with others.

File: iree/compiler/Conversion/LinalgToSPIRV/ConvertToGPUPass.cpp [hunk 2]
```diff
@@ -617,6 +635,9 @@ struct MapLinalgOpToLocalInvocationId : public OpConversionPattern<LinalgOpTy> {
     rewriter.eraseOp(linalgOp);
     return success();
   }
+
+ private:
+  bool optimizeControlFlow;
```
Comment (antiagainst) [inline iree/compiler/Conversion/LinalgToSPIRV/ConvertToGPUPass.cpp:640]: More documentation on what this is for?

**Review by antiagainst — APPROVED** [review]

File: iree/compiler/Conversion/LinalgToSPIRV/ConvertToGPUPass.cpp [hunk 3]
```diff
@@ -532,7 +542,7 @@ struct SerializeParallelLoopPattern
 template <typename LinalgOpTy>
 static LogicalResult mapLinalgOpToLocalInvocationIdImpl(
     LinalgOpTy linalgOp, ArrayRef<Value> operands,
-    ConversionPatternRewriter &rewriter) {
+    ConversionPatternRewriter &rewriter, bool optimizeControlFlow) {
```
Comment (ThomasRaoux) [inline iree/compiler/Conversion/LinalgToSPIRV/ConvertToGPUPass.cpp:545]: Do we need this at all? I thought we didn't want to do this anymore? Does it do anything right now?

**Review by ThomasRaoux — APPROVED** [review]

File: iree/compiler/Conversion/LinalgToSPIRV/ConvertToGPUPass.cpp (hunk 3 above)
Comment (MaheshRavishankar) [reply iree/compiler/Conversion/LinalgToSPIRV/ConvertToGPUPass.cpp:545]: mostly for legacy code. I didnt clean it up, cause this whole pass is probably going to go away.

**Pull request merged**